### DISCUSSION TITLE
Support wiki-link embedding syntax

### DISCUSSION THREAD

**Discussion by kirasok** [zk-org/zk#408] [Ideas]
### Verify

- [X] I searched the existing discussions for similar ideas

### Share your idea or feature request

I use zk for writing and [emanote](https://github.com/srid/emanote) to generate website. 

Emanote supports [wiki-link embedding syntax](https://emanote.srid.ca/guide/markdown/embed) where files (notes, images, video, etc) can be embedded with `![[..]]`. Compared to default markdown, it doesn't specify the path so it doesn't break if the file is moved.

Currently, zk supports notes, but for other media LSP outputs `not found`. I think it would be great to support other filetypes as it would simplify managing of zettelkasten

**Comment by tjex** [comment]
By "support" do you mean not showing a `link not found` error?

Or do you mean to support the viewing of images and video via `zk` within your editor?

Currently this is what we do for images and `file:///` links. We simply ignore these patterns when parsing links within markdown documents.

We could add the `![[link]]` pattern as well, but I'm hesitant to do so as it's non-standard markdown symtax, [see here](https://www.markdownguide.org/basic-syntax/#images-1). 

With more and more flavours of markdown syntax appearing, I'm thinking that it could be valuable to enable users to selectively ignore idiosyncratic markdown syntax from their own configs. Something a long the lines of:

```lua
require("zk").setup({
    -- these links will not show 'not found' errors
    ignore_link_patterns = {"![[]]"} 
})
```

Note to self: perhaps this is better in `config.toml`

**Reply by kirasok** [reply]
I think adding option to ignore link patterns would be perfect
